src/System.Net.Http/src/System/Net/Http/HttpMethod.cs

```diff
@@ -106,7 +122,12 @@ public override bool Equals(object obj)
 
         public override int GetHashCode()
         {
-            return _method.ToUpperInvariant().GetHashCode();
+            if (_hashcode == 0)
+            {
+                _hashcode = _method.ToUpperInvariant().GetHashCode();
```

Comment:
I like this because it keeps the `GetHashCode` optimization within/close-to `GetHashCode()` and doesn't pollute the rest of the code (static fields and private constructor) with hash-code-optimization-related concerns.